### ISSUE TITLE
Hard Prune

### ISSUE THREAD

**Issue by bryanhanson** [cbeleites/hyperSpec#128]
This proposal is an expanded version of the 2nd part of #114.  This issue assumes that the FauxCell data set is accepted (and at that point 114 can be closed).

We also need #126 fixed before proceeding with the below actions.

I am proposing this for several reasons:

* Discussion of the merits of doing this.  This is essentially a "hard prune" to simplify the re-building of `hyperSpec`.  Doing this will leave us with a core `hyperSpec` to which we can then either add stuff back or break-off and re-integrate the proposed peripheral packages.  Maybe this is easier than wrangling the whole enchilada back into full functioning (apologies for the "americanism" of "the whole enchilada", meaning "everything at once").
* The current `Makefile` and associated shuttling of data sets and vignettes is complex, and thus hard to fix.
* Erick and Bryan cannot currently build and check, leaving us unable to contribute fully.  The proposal here is tedious and severe but it would get us back to a functioning workflow, and accomplish several smaller goals along the way.
* The current unit test infrastructure is hacked version of `testthat` -- Claudia built it this way so as to keep the unit tests with the associated functions, a shortcoming of all the available unit test frameworks. There is a big advantage of doing things this way.  But, as `testthat` internals are not officially exposed and subject to change, this is fragile.  `tinytest` also has zero dependencies.

In the following checklist, commit when each task is complete or preferably more often.  I've probably missed something too.

- [ ] Simplify the `Makefile` to accommodate the changes below.
- [ ] Remove all vignettes.
- [ ] Remove all fileio functions.
- [ ] Disable the examples in all `*.R` files.
- [ ] Create the scripts needed to drive unit tests with `tinytest`.
- [ ] Remove the unit tests from `*.R` files and move them into several logically grouped files for use with `tinytest`.
- [ ] Convert the unit tests from `testthat` to `tinytest` syntax, and have them use the `FauxCell` data set.
- [ ] Build, check and repair as needed until the unit tests work.
- [ ] Run `styler`.  Build, check and repair as needed.
- [ ] Convert the examples in the `*.R files to use the `FauxCell` data set.
- [ ] Build, check and repair as needed until the examples run correctly.

While the list is long, none of this is hard. I think two of us can do this in a week w/o too much trouble.

**Comment by cbeleites** [comment]
I'd very much prefer a gradual approach over this *tabula rasa* proposal: in my experience it is too easy to get into a big mess by (temporarily) removing checks. And both examples and vignettes also serve as checks in a certain sense. The more so, as we don't have that many unit tests in place.

I therefore propose to break this into individual issues which we then order according to what should be tackled first, what may be done in parallel etc.

E.g.: 
1. pick one or a few fileio functions to move. 
2. set up specific io package with these functions and example data
3. write unit tests
4. overhaul documentation:
  - [ ] convert to markdown
  - [ ] convert examples that should be unit tests into unit tests
  - [ ] check that suitable example is available
  - [ ] write a short vignette?
5. add new package to fileio `VignetteDepends` (to be discussed -> maybe start separate vignette instead?)
6. remove duplicated stuff from hyperSpec
  - [ ] functions
  - [ ] example files
  - [ ] parts from fileio vignette that are not needed any more.

Rinse & repeat.

**Comment by cbeleites** [comment]
For public record, we discussed @bryanhanson's proposal in a video call and decided:

- to go for gradual, non-breaking changes
- pull the bullet points apart into 
  - those that should go into the `CONTRIBUTIONS.md` guide - even if temporary
  - those that become proper issues (e.g. file import)
  - those that we won't do (now, e.g. changing the unit test framework)

**Comment by bryanhanson** [comment]
This issue/proposal is no longer relevant so closing.
